api/pkg/apis/v1alpha1/model/campaign.go

```diff
@@ -31,7 +34,42 @@ type StageSpec struct {
 	StageSelector string                 `json:"stageSelector,omitempty"`
 	Inputs        map[string]interface{} `json:"inputs,omitempty"`
 	HandleErrors  bool                   `json:"handleErrors,omitempty"`
-	Schedule      *v1alpha2.ScheduleSpec `json:"schedule,omitempty"`
+	Schedule      string                 `json:"schedule,omitempty"`
+}
+
+// UnmarshalJSON customizes the JSON unmarshalling for StageSpec
+func (s *StageSpec) UnmarshalJSON(data []byte) error {
```

Comment:
Just curious when this function is called?

Reply:
during serialization and deserialization

Reply:
Do you mean these marshal and unmarshal functions will be implicitly called whenever objects are marshal and unmarshal?

Reply:
yes, I think so